platform/openide.util/src/org/openide/util/NbCollections.java

```diff
@@ -63,11 +63,9 @@ private NbCollections() {}
      *         to the named type (or they may be null)
      * @throws ClassCastException if some entry in the raw set was not well-typed, and only if <code>strict</code> was true
      */
-    public static <E> Set<E> checkedSetByCopy(Set rawSet, Class<E> type, boolean strict) throws ClassCastException {
-        Set<E> s = new HashSet<E>(rawSet.size() * 4 / 3 + 1);
-        Iterator it = rawSet.iterator();
-        while (it.hasNext()) {
-            Object e = it.next();
+    public static <E> Set<E> checkedSetByCopy(Set<?> rawSet, Class<E> type, boolean strict) throws ClassCastException {
```

Comment:
According to the documentation in this case `rawSet` is intentionally not a generic set. Wouldn't it be better to suppress warning in this case?